collector/receiver/telemetryapireceiver/receiver.go

```diff
@@ -188,100 +200,188 @@ func (r *telemetryAPIReceiver) httpHandler(w http.ResponseWriter, req *http.Requ
-					if logRecord := createReportLogRecord(&scopeLog, record); logRecord != nil {
-						logRecord.Attributes().PutStr("type", el.Type)
-						if t, err := time.Parse(time.RFC3339, el.Time); err == nil {
-							logRecord.SetTimestamp(pcommon.NewTimestampFromTime(t))
-							logRecord.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))
-						}
-					}
+
+				message := createPlatformMessage(requestId, r.faasFunctionVersion, el.Type, record)
+				if message != "" {
+					logRecord.Body().SetStr(message)
 				}
+			} else if line, ok := record["message"].(string); ok {
+				logRecord.Body().SetStr(line)
+			}
+		} else {
+			if r.currentFaasInvocationID != "" {
+				logRecord.Attributes().PutStr(semconv.AttributeFaaSInvocationID, r.currentFaasInvocationID)
+			}
+			// in plain text https://docs.aws.amazon.com/lambda/latest/dg/telemetry-schema-reference.html#telemetry-api-function
+			if line, ok := el.Record.(string); ok {
+				logRecord.Body().SetStr(line)
 			}
 		}
+		if el.Type == string(telemetryapi.PlatformRuntimeDone) {
+			r.currentFaasInvocationID = ""
+		}
 	}
 	return log, nil
 }
 
-// createReportLogRecord creates a log record for the platform.report event
-// returns the log record if successful, otherwise nil
-func createReportLogRecord(scopeLog *plog.ScopeLogs, record map[string]interface{}) *plog.LogRecord {
+func createPlatformMessage(requestId string, functionVersion string, eventType string, record map[string]interface{}) string {
+	switch eventType {
+	case string(telemetryapi.PlatformStart):
+		if requestId != "" && functionVersion != "" {
+			return fmt.Sprintf(platformStartLogFmt, requestId, functionVersion)
+		}
+	case string(telemetryapi.PlatformRuntimeDone):
+		if requestId != "" && functionVersion != "" {
+			return fmt.Sprintf(platformRuntimeDoneLogFmt, requestId, functionVersion)
+		}
+	case string(telemetryapi.PlatformReport):
+		return createPlatformReportMessage(requestId, record)
+	case string(telemetryapi.PlatformInitStart):
+		runtimeVersion, _ := record["runtimeVersion"].(string)
+		runtimeVersionArn, _ := record["runtimeVersionArn"].(string)
+		if runtimeVersion != "" || runtimeVersionArn != "" {
+			return fmt.Sprintf(platformInitStartLogFmt, runtimeVersion, runtimeVersionArn)
+		}
+	case string(telemetryapi.PlatformInitRuntimeDone):
+		status, _ := record["status"].(string)
+		if status != "" {
+			return fmt.Sprintf(platformInitRuntimeDoneLogFmt, status)
+		}
+	case string(telemetryapi.PlatformInitReport):
+		initType, _ := record["initializationType"].(string)
+		phase, _ := record["phase"].(string)
+		status, _ := record["status"].(string)
+		var durationMs float64
+		if metrics, ok := record["metrics"].(map[string]interface{}); ok {
+			durationMs, _ = metrics["durationMs"].(float64)
+		}
+		if initType != "" || phase != "" || status != "" || durationMs != 0 {
+			return fmt.Sprintf(platformInitReportLogFmt, initType, phase, status, durationMs)
+		}
```

Comment:
I'm wondering if it's too defensive maybe, since `durationMs` should never actually be 0. But at the moment we're relying on the default 0 value to determine if the `durationMs` metric was present or not. The following might be clearer, but again maybe too defensive?
```suggestion
		var durationMs float64
		var hasDuration bool
		if metrics, ok := record["metrics"].(map[string]interface{}); ok {
			durationMs, hasDuration = metrics["durationMs"].(float64)
		}
		if initType != "" || phase != "" || status != "" || hasDuration {
			return fmt.Sprintf(platformInitReportLogFmt, initType, phase, status, durationMs)
		}
```